include/abstractpreview.hpp

```diff
@@ -0,0 +1,209 @@
+/*
+ * This file is part of the QPxTool project.
+ * Copyright (C) 2009 Maxim Aldanov <aldmax@mail.ru>, Gennady "ShultZ" Kozlov
+ * <qpxtool@mail.ru>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * See the file "COPYING" for the exact licensing terms.
+ * */
+
+#ifndef QPREVIEW_H
+#define QPREVIEW_H
+
+#include "dpi_metrics.hpp"
+#include <QAbstractScrollArea>
+#include <QEvent>
+#include <QPrinter>
+#include <QVariant>
+//#define PRINTER_CHANGE_DEVICE
+
+class QResizeEvent;
+class QPaintEvent;
+class QAbstractScrollArea;
+class QMouseEvent;
+class QMousePressData;
+class AbstractPreviewPrivate;
+class QVariant;
+class QWheelEvent;
+
+class QPageInfo {
+public:
+  QPageInfo() {
+    fpage = -1;
+    frect = QRect();
+  }
+  QPageInfo(int page, const QRect &r) {
+    fpage = page;
+    frect = r;
```

Comment:
Codacy found an issue: [Variable 'frect' is assigned in constructor body. Consider performing initialization in initialization list.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)